src/wait-strategy.ts

```diff
@@ -74,7 +74,17 @@ export class LogWaitStrategy extends AbstractWaitStrategy {
 
   public async waitUntilReady(container: Dockerode.Container): Promise<void> {
     log.debug(`Waiting for log message "${this.message}"`);
-    const stream = await containerLogs(container);
+    const inspect = await inspectContainer(container);
+
+    // We should not take logs prior to our last finish time into account
+    // Otherwise we risk reporting the container as ready based on old logs
+    // This does not apply to containers that are currently not running (i.e already exited)
+    const finishedAt = inspect.state.finishedAt?.getTime();
+    const logsOptions =
+      finishedAt !== undefined && inspect.state.status === "running"
+        ? { since: finishedAt / 1000, timestamps: true }
+        : undefined;
+    const stream = await containerLogs(container, logsOptions);
```

Comment:
I wonder if we could refactor this a bit, something like:

```typescript
const inspect = await inspectContainer(container);

const logOptions = this.hasContainerRestarted(inspect) 
  ? containerRestartedLogOptions(inspect) 
  : undefined
```

WDYT?

Reply:
Sounds good, will do